website/server/controllers/api-v3/auth.js

```diff
@@ -341,7 +341,15 @@ api.resetPassword = {
     if (validationErrors) throw validationErrors;
 
     const email = req.body.email.toLowerCase();
-    const user = await User.findOne({ 'auth.local.email': email }).exec();
+    const user = await User.findOne({
+      $or: [
+        { 'auth.local.username': email.replace('@', '') },
```

Comment:
is there a risk of this picking up someone who coincidentally has a username that's similar to someone else's email address?